docs/content/guides/6.multistore/3.patterns/2.design/4.different-code-per-store.md

```diff
@@ -1,8 +1,208 @@
+- How to handle breaking changes in component props with step-by-step workflows, including duplicating and modifying files
+- Best practices for managing component markup customization to maintain consistency and avoid duplication
+- How to implement these changes using examples in Next.js and Nuxt to adapt to specific frameworks
+::
+
+## Why Customize Component Markups?
+
+In multibrand setups, different stores often require unique layouts or styling. Customization is necessary to ensure proper alignment with branding and design guidelines while sharing common functionalities across stores. For example:
+
+- Store-specific customizations to match unique branding designs.
+- Adjustments to the structure or layout of a component in one brand without affecting others.
+
+Changing a component’s markup is a **structured process**, and this guide will walk you through **safe modifications** and scenarios where **breaking changes** are unavoidable.
+
+## Safe Markup Changes Without Breaking Props
+
+The most straightforward way to customize a component is by modifying **only its markup** while keeping its props interface unchanged. This ensures compatibility across the codebase and minimizes any disruption to dependent components or features.
+
+### When to Use
+
+- To rearrange the component structure.
+- To adjust styling for specific stores or brands without altering the functionality.
+
+### The Solution: Component Override
+
+In cases where a specific store requires customization, you can override the component by duplicating and modifying it within that store's directory.
+
+#### Example Workflow
+
+1. **Duplicate** the component inside the store-specific directory.
+2. **Modify** the duplicated component as per the store's requirements (e.g., layout, structure, or styles).
+
+#### In Next.js
+
+Here’s how to override `ExampleComponent` for a specific store (`store-a`) in a Next.js project:
+
+```tsx
+// /apps/stores/store-a/storefront-unified-nextjs/components/example-component.tsx
+
+interface ExampleComponentProps {
+  title: string;
+  imageSrc: string;
+}
+
+export default function ExampleComponent({ title, imageSrc }: ExampleComponentProps) {
+  return (
+    <div className="store-a example-component"> {/* Customized className for store-a */}
+      <img src={imageSrc} alt={title} />
+      <h2>{title}</h2>
+      {/* Customized content for store-a */}
+    </div>
+  );
+}
+```
+
+#### In Nuxt
+
+The process is similar in Nuxt. Here’s an example:
+
+```vue
+// /apps/stores/store-a/storefront-unified-nuxt/components/example-component.vue
+
+<template>
+  <!-- Customized className for store-a -->
+  <div class="store-a example-component">
+    <img :src="imageSrc" :alt="title" />
+    <h2>{{ title }}</h2>
+    <!-- Customized content for store-a -->
+  </div>
+</template>
+
+<script setup>
+defineProps({
+  title: {
+    type: String,
+    required: true,
+  },
+  imageSrc: {
+    type: String,
+    required: true,
+  },
+});
```

Comment:
props are usually defined in `types.ts` file

Reply:
Yup, i just wanted the example to be simpler one and i've put it into single file.

Reply:
ok, so if so, we could at least use type-based declaration, to follow the same conventions as we do in our code https://vuejs.org/guide/typescript/composition-api.html#typing-component-props

Reply:
Sure, done